File: src/main/java/tlint/TLintProjectComponent.java
package tlint;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationType;
import com.intellij.notification.Notifications;
import com.intellij.openapi.components.ProjectComponent;
import org.jetbrains.annotations.NotNull;

public class TLintProjectComponent implements ProjectComponent {
    public String lintExecutable = System.getProperty("user.home") + "/.composer/vendor/bin/tlint";

    private static final String PLUGIN_NAME = "TLint plugin";

    @NotNull
    @Override
    public String getComponentName() {
        return "TLintProjectComponent";
    }

    public static void showNotification(String content, NotificationType type) {
        Notification errorNotification = new Notification(PLUGIN_NAME, PLUGIN_NAME, content, type);
        Notifications.Bus.notify(errorNotification);
    }
}

File: src/main/kotlin/tlint/TLintProjectComponent.kt
package tlint

import com.intellij.notification.Notification
import com.intellij.notification.NotificationType
import com.intellij.notification.Notifications
import com.intellij.openapi.components.ProjectComponent

class TLintProjectComponent : ProjectComponent {
    override fun getComponentName(): String {
        return "TLintProjectComponent"
    }

    companion object {
        private const val PLUGIN_NAME = "TLint plugin"

        fun showNotification(content: String, type: NotificationType) {
            val errorNotification = Notification(PLUGIN_NAME, PLUGIN_NAME, content, type)
            Notifications.Bus.notify(errorNotification)
        }
    }
}